### DISCUSSION TITLE
Clarify the relationship between the docker and dagger images cache

### DISCUSSION THREAD

**Discussion by marcosnils** [dagger/dagger#12514] [Ideas]
When `docker build` is used with `buildkit` mode in the same machine, images present through `docker pull` are shared with buildkit automatically (not sure how it actually works under the hood). Some users (https://discord.com/channels/707636530424053791/1073696894380949545) which are currently trying out dagger pre-pull docker images as part of their node bootstrap and would like those images to be present in their Dagger pipelines as well so they don't have to be re-downloaded.

Probably @aluzzardi and @sipsma know a bit more about this.

**Comment by sipsma** [comment]
> When docker build is used with buildkit mode in the same machine, images present through docker pull are shared with buildkit automatically (not sure how it actually works under the hood)

This works because `docker` doesn't use vanilla buildkit, it imports buildkit as a library and then plugs in its own snapshotter and worker backend that integrates with docker's content stores: https://github.com/moby/moby/tree/master/builder

In docker's most recent release I believe support has been added for using containerd snapshotter+content-store backends (? not 100% sure) so this might have been simplified even further now.

Either way, `dagger` can't use that "moby-flavored" buildkit out-of-the-box since it is specialized for moby and we require our "dagger-flavored" buildkit that comes with extras like the dagger shim.

---

Hypothetically, if docker does actually support containerd snapshotter and content-store backends now, it might be possible to support our dagger buildkit engine to use that same containerd worker backend, which *might* enable more seamless sharing. No clue if that would really work or be worth it though, would require a lot of investigation.
